core/src/main/java/com/es/core/service/impl/HttpSessionCartService.java

```diff
@@ -1,30 +1,100 @@
 package com.es.core.service.impl;
 
+import com.es.core.dao.PhoneDao;
+import com.es.core.dao.StockDao;
+import com.es.core.exception.EntityNotFoundException;
+import com.es.core.exception.OutOfStockException;
 import com.es.core.model.cart.Cart;
+import com.es.core.model.cart.CartItem;
+import com.es.core.model.phone.Phone;
+import com.es.core.model.phone.Stock;
 import com.es.core.service.CartService;
 import org.springframework.stereotype.Service;
 
+import javax.annotation.Resource;
+import java.math.BigDecimal;
 import java.util.Map;
+import java.util.Optional;
 
 @Service
 public class HttpSessionCartService implements CartService {
+    @Resource
+    private Cart cart;
+    @Resource
+    private PhoneDao phoneDao;
+    @Resource
+    private StockDao stockDao;
+
     @Override
     public Cart getCart() {
-        throw new UnsupportedOperationException("TODO");
+        return cart;
     }
 
     @Override
     public void addPhone(Long phoneId, Long quantity) {
-        throw new UnsupportedOperationException("TODO");
+        Phone phone = phoneDao.get(phoneId).orElseThrow(() -> new EntityNotFoundException("Phone", phoneId));
+
+        Long inCartQuantity = findCartItem(phoneId)
+                .map(CartItem::getQuantity)
+                .orElse(0L);
+
+        tryToAddItemToCart(phone, quantity, inCartQuantity);
     }
 
     @Override
     public void update(Map<Long, Long> items) {
-        throw new UnsupportedOperationException("TODO");
+        for (Map.Entry<Long, Long> entry : items.entrySet()) {
+            Long phoneId = entry.getKey();
+            Long quantity = entry.getValue();
+
+            Phone phone = phoneDao.get(phoneId).orElseThrow(() -> new EntityNotFoundException("Phone", phoneId));
+            tryToAddItemToCart(phone, quantity, 0L);
+        }
     }
 
     @Override
     public void remove(Long phoneId) {
-        throw new UnsupportedOperationException("TODO");
+        cart.removeItem(phoneId);
+        recalculateTotalQuantityAndPrice();
+    }
+
+    private Optional<CartItem> findCartItem(Long phoneId) {
+        return Optional.ofNullable(cart.getItems().get(phoneId));
+    }
+
+    private void tryToAddItemToCart(Phone phone, Long quantity, Long inCartQuantity) {
```

Comment:
bad name, do not use try, better to name addProductToCart, "try" will be "covered" with thrown exception